cryo_utilities.sh

```diff
@@ -6,111 +6,111 @@
 hasPass=$(passwd -S "$USER" | awk -F " " '{print $2}')
 if [[ ! $hasPass == "P" ]]; then
     zenity --error --title="Password Error" --text="Password is not set, please set one in the terminal with the <b>passwd</b> command, then run this again." --width=400 2> /dev/null
-else
-    PASSWD="$(zenity --password --title="Enter Password" --text="Enter Deck User Password (not Steam account!)" 2>/dev/null)"
-    echo "$PASSWD" | sudo -v -S
-    ans=$?
-    if [[ $ans == 1 ]]; then
-        zenity --error --title="Password Error" --text="Incorrect password provided, please run this command again and provide the correct password." --width=400 2> /dev/null
+    exit 1
+fi
+PASSWD="$(zenity --password --title="Enter Password" --text="Enter Deck User Password (not Steam account!)" 2>/dev/null)"
+echo "$PASSWD" | sudo -v -S
+ans=$?
+if [[ $ans == 1 ]]; then
+    zenity --error --title="Password Error" --text="Incorrect password provided, please run this command again and provide the correct password." --width=400 2> /dev/null
+    exit 1
+fi
+# Thank you to ssorgatem for the dynamic swapfile location code.
+# Note: The swapfile recreation does not currently support BTRFS
+SWAPFILE=$(swapon | grep swapfile | cut -d" " -f1)
+if [ ! -f "$SWAPFILE" ]; then
+    echo "Swapfile does not exist, generating a default-sized file where it should be located..."
+    sudo dd if=/dev/zero of=/home/swapfile bs=1G count=1 status=none
+    sudo chmod 0600 /home/swapfile
+    sudo mkswap /home/swapfile
+    sudo swapon /home/swapfile
+    SWAPFILE="/home/swapfile"
+fi
+if zenity --question --title="Disclaimer" --text="This script was made by CryoByte33 to resize the swapfile on a Steam Deck.\n\n<b>Disclaimer: I am in no way responsible to damage done to any device this is executed on, all liability lies with the runner.</b>\n\nDo you accept these terms?" --width=600 2> /dev/null; then
```

Comment:
Building upon your approach, shouldn't we move this to the top of the script and make it exit immediately e.g.

```bash
# !/usr/bin/bash

if !(zenity --question --title="Disclaimer" --text="This script was made by CryoByte33 to resize the swapfile on a Steam Deck.\n\n<b>Disclaimer: I am in no way responsible to damage done to any device this is executed on, all liability lies with the runner.</b>\n\nDo you accept these terms?" --width=600 2> /dev/null); then
        zenity --error --title="Terms Denied" --text="Terms were denied, cannot proceed." --width=300 2> /dev/null
        exit 1
fi

...
```

IMO, this would be better from a design standpoint too (requires the terms to be accepted before anything actions).

Reply:
This is also a good call, I'll accept it as well and do one final test 🙂